flightreservation/Views/SignUp.js

```diff
@@ -1,35 +1,167 @@
+import Checkbox from '@react-native-community/checkbox';
+import Icon from 'react-native-vector-icons/AntDesign';
 
-import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
+import auth from '@react-native-firebase/auth';
+
+import firestore from '@react-native-firebase/firestore';
+
+const SignUp = ({ navigation }) => {
+  const [firstNameInput, setFirstNameInput] = useState(false);
+  const [emailInput, setEmailInput] = useState(false);
+  const [passwordInput, setPasswordInput] = useState(false);
+  const [showPassword, setShowPassword] = useState(true);
+  const [toggleCheckBox1, setToggleCheckBox1] = useState(false);
+  const [toggleCheckBox2, setToggleCheckBox2] = useState(false);
+  const allValid = (firstNameInput && emailInput && passwordInput && toggleCheckBox1) ? true : false
+  const [errorEmail, setErrorEmail] = useState('');
+  const [errorPassword, setErrorPassword] = useState('');
 
-const SignUp = ({navigation}) => {
   return (
     <>
       <View style={styles.container}>
         <Text style={styles.header}>Sign Up</Text>
         <View style={styles.containerSpacing}>
-          <UserInfo />
+          <View>
+            <Text style={styles.label}>First Name</Text>
+            <TextInput
+              style={[styles.input, { borderColor: firstNameInput ? 'blue' : 'black' }]}
+              onChangeText={(value) => setFirstNameInput(value)}
+            />
+            <View style={{ flexDirection: 'row' }}>
+              <Text style={styles.label}>Email</Text>
+              <Text style={{ color: errorEmail != '' ? 'red' : 'black' }}> *<Text>{errorEmail}</Text></Text>
+            </View>
+            <TextInput
+              style={[styles.input, { borderColor: emailInput ? 'blue' : 'black' }]}
+              keyboardType="email-address"
+              onChangeText={(value) => setEmailInput(value)}
+            />
+            <View style={{ flexDirection: 'row' }}>
+              <Text style={styles.label}>Password</Text>
+              <Text style={{ color: errorPassword != '' ? 'red' : 'black' }}> *<Text>{errorPassword}</Text></Text>
+            </View>
+            <View
+              style={[
+                styles.input,
+                styles.passwordContainer,
+                { borderColor: passwordInput ? 'blue' : 'black' },
+              ]}>
+              <TextInput
+                style={styles.passwordInput}
+                secureTextEntry={showPassword}
+                onChangeText={(value) => setPasswordInput(value)}
+              />
+              <Icon
+                style={styles.eye}
+                name="eye"
+                color={passwordInput ? '#5C6EF8' : '#A9A9A9'}
+                size={25}
+                onPress={() => setShowPassword(!showPassword)}
+              />
+            </View>
+            <Text style={styles.textFeedback}>
+              Use 8 or more characters with a mix of letter, numbers and symbols.
```

Comment:
as my partner said, you better have a file of constants for both colors and texts